docs/development/out-of-tree.md

```diff
@@ -0,0 +1,97 @@
+(out-of-tree)=
+
+# Building and testing Python packages out of tree
+
+This is some information about how to build and test Python packages against
+Pyodide out of tree (for instance in your package's CI or for use with private
+packages).
+
+## Building binary packages for Pyodide
+
+If your package is a pure Python package (i.e., if the wheel ends in
+`py3-none-any.whl`) then it can be built in whichever way you normally build it.
+Otherwise, the procedure is simple. In your package directory run the following
+command line commands:
+
+```sh
+pip install pyodide-build
+pyodide build
```

Comment:
Also, where does it specify which version of Pyodide it's going to use? Is there a mapping between pyodide-build and pyodide versions somewhere?

Reply:
By default it uses the same version as `pyodide-build`. We really ought to add a separate command to install the xbuildenv rather than doing it automatically.